astropy/tests/helper.py

```diff
@@ -460,53 +459,4 @@ def assert_quantity_allclose(actual, desired, rtol=1.e-7, atol=None,
     This is a :class:`~astropy.units.Quantity`-aware version of
     :func:`numpy.testing.assert_allclose`.
     """
-    import numpy as np
-    np.testing.assert_allclose(*_unquantify_allclose_arguments(actual, desired,
-                                                               rtol, atol),
-                               **kwargs)
-
-
-def quantity_allclose(a, b, rtol=1.e-5, atol=None, **kwargs):
-    """
-    Returns True if two arrays are element-wise equal within a tolerance.
-
-    This is a :class:`~astropy.units.Quantity`-aware version of
-    :func:`numpy.allclose`.
-    """
-    import numpy as np
-    return np.allclose(*_unquantify_allclose_arguments(a, b, rtol, atol),
-                       **kwargs)
-
-
-def _unquantify_allclose_arguments(actual, desired, rtol, atol):
-    from .. import units as u
-
-    actual = u.Quantity(actual, subok=True, copy=False)
-
-    desired = u.Quantity(desired, subok=True, copy=False)
-    try:
-        desired = desired.to(actual.unit)
-    except u.UnitsError:
-        raise u.UnitsError("Units for 'desired' ({0}) and 'actual' ({1}) "
-                           "are not convertible"
-                           .format(desired.unit, actual.unit))
-
-    if atol is None:
-        # by default, we assume an absolute tolerance of 0
-        atol = u.Quantity(0)
-    else:
-        atol = u.Quantity(atol, subok=True, copy=False)
-        try:
-            atol = atol.to(actual.unit)
-        except u.UnitsError:
-            raise u.UnitsError("Units for 'atol' ({0}) and 'actual' ({1}) "
-                               "are not convertible"
-                               .format(atol.unit, actual.unit))
-
-    rtol = u.Quantity(rtol, subok=True, copy=False)
-    try:
-        rtol = rtol.to(u.dimensionless_unscaled)
-    except Exception:
-        raise u.UnitsError("`rtol` should be dimensionless")
-
-    return actual.value, desired.value, rtol.value, atol.value
+    assert quantity_allclose(actual, desired, rtol, atol, **kwargs)
```

Comment:
I was about to send the following comment on this suggestion by @astrofrog when the changes were pushed :-)... not sure I agree:
"""
Before we make that change: `assert_allclose` prints out some specifics about what is not matching, which `assert quantity_allclose` would not do.

One solution to importing a private function might be to not make it private (but not list it in `__all__` in `quantity.py` (or perhaps move it to `quantity_helper`).

But perhaps best would be to have a public helper function (or decorator?) a bit like `quantity_input` that allows one to pass in that the units of some arguments should be the consistent.
"""
For now, my suggestion would be to keep using `_unquantify_allclose_arguments`, and keep the wish for a new function separate -  See #7368.